Add resource pack stuff

Blueyescat opened the pull request:
### Description
Adds an effect to send a resource pack to players, an event to listen to resource pack request responses, a condition to check the response in the event, an expression to get the last resource pack response of a player, a condition to check whether a player has a server resource pack loaded.

**Test Script**
```vb
#!send resource pack "https://drive.google.com/uc?export=download&id=1vvkWrR7LcGyveExwKchiSn6PAc3qVS-z" to player
on resource pack request response:
	send "%player's resource pack action%"
	if the resource pack was accepted:
		send "thanks"
	else if the resource pack was declined or failed to download:
		send "isn't loaded"
	if the resource pack wasn't accepted or success:
		send "^ the same"

on resource pack accept:
	send "Accepted"

on resource pack decline:
	send "Declined"

on resource pack success:
	send "Successfully Loaded"

on resource pack fail:
	send "Download Failed"

on resource pack decline or fail:
	send "Declined or failed"
```

---
**Target Minecraft Versions:** any
**Requirements:** Paper is needed for ExprLastResourcePackResponse and CondHasResourcePack only
**Related Issues:** None

Nicofisi commented on src/main/java/ch/njol/skript/events/EvtResourcePackAction.java:
```diff
@@ -0,0 +1,83 @@
+						"and once when the resource pack is successfully installed or failed.")
+				.examples("on resource pack request action:",
+						"	if the action is decline or download fail:",
+						"",
```
Might want to put something here maybe?

Blueyescat commented:
@Nicofisi yeah `on resource pack something or something else:` works fine, already it is in the test script in the issue. Thanks 🙂

Blueyescat commented:
As discussed in Discord, i used "state" and "response" instead of "action", removed the global event action expression and added a condition to check the response instead. I have updated the test script in the issue.

Blueyescat commented on src/main/java/ch/njol/skript/conditions/CondResourcePack.java:
```diff
@@ -0,0 +1,80 @@
+	
+	@Override
+	public String toString(final @Nullable Event e, final boolean debug) {
+		return "resource pack was " + (isNegated() ? " not" : "") + states.toString(e, debug);
```
Good catch :+1:
